pydis_site/apps/resources/templatetags/as_css_class.py

```diff
@@ -0,0 +1,18 @@
+from django import template
+
+register = template.Library()
+
+
+@register.filter
+def as_css_class(class_name: str) -> str:
+    """
+    Convert any string to a css-class name.
+
+    For example, convert
+    "Favorite FROOT_is_LEMON" to
+    "favorite-froot-is-lemon"
+    """
```

Comment:
This is called `kebab-case`, I learned today. Let's rename this function to use the correct term, and maybe cover more edge cases, too.